### PR TITLE
Squad Identification Fluff and Helmet Garb Buff

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Adds squad-colored scarves and headbands.
Adds a third slot to helmets that only accepts fluff helmet garb.

Additionally, expanded the helmet garb system to have object flags that determine whether the item needs a helmet band, and whether it should be considered a garb item for the garb slot.

## Why It's Good For The Game

Allows for more marine customization via the headbands and scarves, and also reinforces squad identity by making them match the squad's color.

## Changelog

:cl: Geeves, Frans_Feiffer
add: Added squad-colored scarves and headbands.
add: Added a third slot to helmets that only takes fluff helmet garb.
/:cl:

### CODE REVIEW

**Comment by Geevies** [comment]
still waiting on a sprite or two

File: code/game/objects/items/storage/internal.dm [hunk 1]
```diff
@@ -166,3 +166,29 @@
...
 	. = ..()
 	master_object = null
+
+
+// Marine Helmet Storage
+/obj/item/storage/internal/helmet
+	var/list/garb_items
+	var/slots_reserved_for_garb
+
+/obj/item/storage/internal/helmet/can_be_inserted(obj/item/W, stop_messages) //We don't need to stop messages, but it can be left in.
```
Comment (stanalbatross) [inline code/game/objects/items/storage/internal.dm:176]: Var name!

File: code/game/objects/items/storage/internal.dm [hunk 2]
```diff
@@ -166,3 +166,29 @@
...
+	. = ..()
+	if(!.)
+		return
+
+	if(!HAS_FLAG(W.flags_obj, OBJ_IS_HELMET_GARB) && length(contents) - length(garb_items) >= storage_slots - slots_reserved_for_garb)
+		if(!stop_messages)
+			to_chat(usr, SPAN_WARNING("This slot is reserved for helmet accessories!"))
+		return FALSE
+
+/obj/item/storage/internal/helmet/_item_insertion(obj/item/W, prevent_warning = 0)
```
Comment (stanalbatross) [inline code/game/objects/items/storage/internal.dm:186]: Var name also if arg is boolean use truefalse

File: code/game/objects/items/storage/internal.dm [hunk 3]
```diff
@@ -166,3 +166,29 @@
...
+	if(HAS_FLAG(W.flags_obj, OBJ_IS_HELMET_GARB))
+		LAZYADD(garb_items, W)
+	return ..()
+
+/obj/item/storage/internal/helmet/_item_removal(obj/item/W, atom/new_location)
```
Comment (stanalbatross) [inline code/game/objects/items/storage/internal.dm:191]: Var name

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]
code looks good, soul verified

**Review by nauticall — APPROVED** [review]
sprites look goooood

**Pull request merged**